core/rag/simple_rag.py

```
@@ -102,18 +102,70 @@ def _score(query: str, text: str) -> float:
+    """Get RAG system statistics."""
+    items = _get_index()
+    total_chunks = len(items)
+    sources = {}
+    for src, _ in items:
+        src_name = Path(src).name
+        sources[src_name] = sources.get(src_name, 0) + 1
+
+    return {"total_chunks": total_chunks, "sources": sources, "index_loaded": _INDEX is not None}
+
+
+def retrieve_context(query: str, max_chunks: int = 3) -> str:
```

Ruff (0.13.3)

Unused function argument: `query`

(ARG001)